python/dolfinx/wrappers/mesh.cpp

```diff
@@ -167,6 +167,28 @@ void declare_mesh(nb::module_& m, std::string type)
   std::string pyclass_geometry_name = std::string("Geometry_") + type;
   nb::class_<dolfinx::mesh::Geometry<T>>(m, pyclass_geometry_name.c_str(),
                                          "Geometry object")
+      .def(
+          "__init__",
+          [](dolfinx::mesh::Geometry<T>* self,
+             std::shared_ptr<const dolfinx::common::IndexMap> index_map,
+             nb::ndarray<std::int32_t, nb::ndim<1>, nb::c_contig> dofmap,
+             const dolfinx::fem::CoordinateElement<T>& element,
+             nb::ndarray<T, nb::ndim<1>, nb::c_contig> x, int dim,
```

Comment:
Make `x` 2D and remove `dim`.

Reply:
I don't think we can do that, as we send in 3D coordinates, even when we want to use 1D or 2D meshes.

Reply:
Send in 1D, 2D or 3D coords. Simpler and no significant performance issue since the data gets copied anyway.